bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/dto/ChannelDTO.java

```diff
@@ -48,6 +49,7 @@ public ChannelDTO(ChannelUID uid, String channelTypeUID, String itemType, String
         this.properties = properties;
         this.configuration = toMap(configuration);
         this.defaultTags = new HashSet<>(defaultTags);
+        this.kind = kind.toString();
```

Comment:
How are existing things created? I added a `kind = STATE` to every `Channel` constructor, so that should never be null.